src/battle_ai_util.c

```diff
@@ -753,6 +770,25 @@ static bool32 AI_IsMoveEffectInMinus(u32 battlerAtk, u32 battlerDef, u32 move, s
                     break;
                 case MOVE_EFFECT_RECHARGE:
                     return gMovesInfo[move].additionalEffects[i].self;
+                case MOVE_EFFECT_ATK_PLUS_1:
+                case MOVE_EFFECT_DEF_PLUS_1:
+                case MOVE_EFFECT_SPD_PLUS_1:
+                case MOVE_EFFECT_SP_ATK_PLUS_1:
+                case MOVE_EFFECT_SP_DEF_PLUS_1:
+                case MOVE_EFFECT_EVS_PLUS_1:
+                case MOVE_EFFECT_ACC_PLUS_1:
+                case MOVE_EFFECT_ATK_PLUS_2:
+                case MOVE_EFFECT_DEF_PLUS_2:
+                case MOVE_EFFECT_SPD_PLUS_2:
+                case MOVE_EFFECT_SP_ATK_PLUS_2:
+                case MOVE_EFFECT_SP_DEF_PLUS_2:
+                case MOVE_EFFECT_EVS_PLUS_2:
+                case MOVE_EFFECT_ACC_PLUS_2:
+                case MOVE_EFFECT_ALL_STATS_UP:
+                    if ((gMovesInfo[move].additionalEffects[i].self && GetBattlerAbility(battlerAtk) == ABILITY_CONTRARY)
+                        || (noOfHitsToKo != 1 && abilityDef != ABILITY_CONTRARY && !IsMoldBreakerTypeAbility(abilityAtk)))
```

Comment:
I think the check for battleDef is unnecessary in this place. 

Also not related to the pr and can be changed later but for ai calcs it is better to use stored values so in this case `GetBattlerAbility(battlerAtk)` should be replaced with `AI_DATA->abilities{battlerAtk}`

Reply:
I believe it's unnecessary right now, but necessary for custom moves or future moves.
The mold breaker check is a mistake however.

Reply:
could you give an example for a scenario where it could be needed?

Reply:
For a move that deals damage and raises the target's attack.

Reply:
I see. That's fine then.